core-bundle/src/String/HtmlAttributes.php

```diff
@@ -0,0 +1,248 @@
+     */
+    public static function fromString(string $attributesString): self
+    {
+        $attributes = [];
+
+        $length = \strlen($attributesString);
+        $currentName = $currentValue = null;
+        $readName = $hasDelimiter = true;
+
+        for ($position = 0; $position < $length; ++$position) {
+            // Advance position until the next character is no whitespace and return that character
+            $advance = static function () use ($attributesString, $length, &$position): string {
+                $next = '';
+
+                for ($i = $position + 1; $i < $length; ++$i) {
+                    if (($next = $attributesString[$i]) !== ' ') {
+                        break;
+                    }
+                }
+
+                $position = $i - 1;
+
+                return $next;
+            };
+
+            $currentChar = $attributesString[$position];
+            $isWhitespace = ' ' === $currentChar;
+
+            // Ignore whitespaces between names and values
+            if ($isWhitespace && (($readName && null === $currentName) || (!$readName && null === $currentValue))) {
+                continue;
+            }
+
+            // Parse name
+            if ($readName) {
+                if ($isWhitespace) {
+                    if ('=' !== $advance()) {
+                        $attributes[$currentName] = null;
+                        $currentName = '';
+                    }
+                } elseif ('=' === $currentChar) {
+                    $readName = false;
+
+                    if ('"' !== $advance()) {
+                        $hasDelimiter = false;
+                    } else {
+                        ++$position;
+                    }
+                } else {
+                    $currentName .= $currentChar;
+                }
+
+                continue;
+            }
+
+            // Parse value
+            if ('"' === $currentChar || (!$hasDelimiter && $isWhitespace)) {
+                $attributes[$currentName] = $currentValue;
+
+                $currentName = $currentValue = null;
+                $readName = $hasDelimiter = true;
+            } else {
+                $currentValue .= $currentChar;
+            }
+        }
+
+        // Add remaining value
+        if (null !== $currentValue) {
+            $attributes[$currentName] = $currentValue;
+        }
+
+        return new self($attributes);
+    }
+
+    public function set(string $name, string|int|bool|\Stringable|null $value): self
+    {
+        $name = strtolower($name);
+
+        if (1 !== preg_match('/^[a-z](?:-?[a-z0-9])*$/', $name)) {
+            throw new \InvalidArgumentException(sprintf('A HTML attribute name must only consist of the characters [a-z0-9-], must start with a letter, must not end with a hyphen and must not contain two hyphens in a row, got "%s".', $name));
+        }
+
+        $this->attributes[$name] = null !== $value ? (string) $value : null;
```

Comment:
```suggestion
        if (false === $value) {
            unset($this->attributes[$name]);
        } else {
            $this->attributes[$name] = true === $value ? '' : (string) $value;
        }
```

As in HTML, boolean attributes are set by existing or not existing, we should unset `false` values IMO.

This way something like this should work as intended:

```twig
<video{{ html_attributes({ src: 'video.mp4', loop: true, muted: false, controls: true}) }}>
```

Reply:
see 545992fbdb5b7cc859fa9181e829ae37ecc3517f